api/Controllers/AccountController.cs

```diff
@@ -0,0 +1,100 @@
+using api.Models;
+using api.Dto.Account;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using api.Data;
+
+namespace api.Controllers
+{
+    [Route("api/account")]
+    [ApiController]
+    public class AccountController : ControllerBase
+    {
+        private readonly ApiDBContext _dbContext;
+        private readonly ITokenService _tokenService;
+        private readonly IStudentRepository _studentManager;
+        private readonly IPasswordHasher<Student> _passwordHasher;
+
+        public AccountController(ITokenService tokenService, IPasswordHasher<Student> passwordHasher, IStudentRepository studentManager, ApiDBContext dbContext)
+        {
+            _tokenService = tokenService;
+            _passwordHasher = passwordHasher;
+            _studentManager = studentManager;
+            _dbContext = dbContext;
+        }
+
+        [HttpPost("login")]
+        public async Task<IActionResult> Login(LoginDto loginDto)
+        {
+            if (!ModelState.IsValid) {
+                return BadRequest(ModelState);
+            }
+
+            var user = await _dbContext.Students.FirstOrDefaultAsync(x => x.StudentId == loginDto.StudentId);
+
+            if (user == null) return NotFound("Неверный номер студенческого билета!");
+
+            var result  = _passwordHasher.VerifyHashedPassword(user, user.PasswordHash, loginDto.Password);
+
+            if (result != PasswordVerificationResult.Success) return NotFound("Неверное имя пользователя или пароль!");
+
+            try {
+                return Ok(
+                    new UserDto
+                    {
+                        StudentId = user.StudentId,
+                        Token = _tokenService.CreateToken(user)
+                    }
+                );
+            } catch (Exception e) {
+                return BadRequest(e.Message);
+            }
+        }
+
+        [HttpPost("register")]
+        public async Task<IActionResult> Register([FromBody] RegisterDto registerDto)
+        {
+            try
+            {
+                if (!ModelState.IsValid) {
+                    return BadRequest(ModelState);
+                }   
+
+                var existingStudent = await _dbContext.Students.FirstOrDefaultAsync(x => x.StudentId == registerDto.StudentId);
+
+                if (existingStudent != null) {
+                    return BadRequest("Such a user already exists!");
+                }
+
+                var passwordHash = _passwordHasher.HashPassword(null!, registerDto.Password);
+
+                var student = new Student
+                {
+                    StudentId = registerDto.StudentId,
+                    Group = registerDto.Group,
+                    FullName = registerDto.FullName,
+                    Email = registerDto.Email,
+                    PasswordHash = passwordHash
+                };
+
+                var createUser = await _studentManager.CreateAsync(student);
```

Comment:
Наверное пользователь уже создан к тому моменту)

```suggestion
                var createdUser = await _studentManager.CreateAsync(student);
```

Reply:
Ну по сути пользователя мы создаем именно методом CreateAsync

Reply:
Так ты в переменную записываешь уже созданного пользователя. Так что, ты его уже создал ранее :)

Reply:
Типа именования корректные в формате:

- `currentTime`
- `createdUser`
- `isValidData`
- `shouldShowToast`